test/extended/baremetal/high_availability.go

```diff
@@ -0,0 +1,117 @@
+package baremetal
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	interval           = 10 * time.Second
+	waitTimeout        = 5 * time.Minute
+	baremetalNamespace = "openshift-machine-api"
+)
+
+var _ = g.Describe("[sig-installer][Feature:baremetal][Serial] Baremetal platform should ensure", func() {
+	defer g.GinkgoRecover()
+
+	oc := exutil.NewCLI("baremetal")
+
+	g.It("cluster baremetal operator returns back healthy after it is deleted", func() {
+		skipIfNotBaremetal(oc)
+
+		c, err := e2e.LoadClientset()
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		g.By("delete cluster baremetal operator")
+		err = c.AppsV1().Deployments(baremetalNamespace).Delete(context.Background(), "cluster-baremetal-operator", v1.DeleteOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("wait until cluster baremetal operator returns back healthy")
+		err = wait.Poll(interval, waitTimeout, func() (bool, error) {
+			dc, err := c.AppsV1().Deployments(baremetalNamespace).Get(context.Background(), "cluster-baremetal-operator", v1.GetOptions{})
+			if err != nil {
+				// This error is expected at first run
+				if strings.Contains(err.Error(), `deployments.apps "cluster-baremetal-operator" not found`) {
+					return false, nil
+				}
+				return false, err
+			}
+
+			if dc.Status.AvailableReplicas != 1 {
+				return false, nil
+			}
+
+			return true, nil
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("verify that baremetal hosts are healthy and correct state")
+		checkMetal3DeploymentHealthy(oc)
+	})
+
+	g.It("metal3 deployment returns back healthy after it is deleted", func() {
+		skipIfNotBaremetal(oc)
+
+		c, err := e2e.LoadClientset()
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		g.By("delete metal3 deployment")
+		err = c.AppsV1().Deployments(baremetalNamespace).Delete(context.Background(), "metal3", v1.DeleteOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("wait until metal3 deployment returns back healthy")
+		err = wait.Poll(interval, waitTimeout, func() (bool, error) {
+			dc, err := c.AppsV1().Deployments(baremetalNamespace).Get(context.Background(), "metal3", v1.GetOptions{})
+			if err != nil {
+				// This error is expected at first run
+				if strings.Contains(err.Error(), `deployments.apps "metal3" not found`) {
+					return false, nil
+				}
+				return false, err
+			}
+
+			if dc.Status.AvailableReplicas != 1 {
+				return false, nil
+			}
+
+			return true, nil
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("verify that baremetal hosts are healthy and correct state")
+		checkMetal3DeploymentHealthy(oc)
```

Comment:
Also in this case I don't think the resources are affected after the deletion? In such cases a waiting condition could be required instead of an immediate check

Reply:
I deliberately did not wrap this with wait condition, because metal3 deployments should not be affected by this deletion.